### ISSUE TITLE
why do not use https://github.com/chalk/chalk

### ISSUE THREAD

**Issue by postbird** [Marak/colors.js#293]
why do not use https://github.com/chalk/chalk

**Comment by panoply** [comment]
Nah, not chalk. Sindre needs humbling and ESM is not a viable option yet. Look for alternatives here:

- https://github.com/jorgebucaran/colorette
- https://github.com/lukeed/kleur

Also, Marak don't owe anyone anything. Too many devs out here spitting venom about what he is doing are the same folks who don't know what it's like to have nothing, they come from money and are silver spooned. Far too many sitting in the ivory tower taking checks from the same companies making millions from OS and folks like Marak. These are his repos and projects. The man needs bread and he needs to eat. He within in right to do this. Hopefully it invokes change. @Marak don't listen to the pronouns, a lot of us f*ck with you, you a real one.

**Comment by dnlsndr** [comment]
@panoply Couldn't have said it better. It's @Maraks right to do what he did. Anybody think AWS ever donated money to the colors.js project since they use it for their AWS CLI?! Nope...

**Comment by postbird** [comment]
> 



> @panoply Couldn't have said it better. It's @maraks right to do what he did. Anybody think AWS ever donated money to the colors.js project since they use it for their AWS CLI?! Nope...

I have no right to accuse @maraks of any of its practices. The main reason I use `chalk` is that we can avoid these potential risks by choosing items that are more standardized, more stable, and reviewed by multiple people.

Of course, I'll reserve my opinion, @maraks actions also hurt thousands of people who work for open source software. This kind of behavior will not change the behavior of those big companies like AWS, they have strong R&D capabilities, and "use doctrine" is just an incidental behavior. Even if we all shut down all open source software, it won't prompt them to donate. In the end it's still the people who participate and contribute to open source that get hurt.

Also, I think if the original intention of open source is for money, then why not sell closed source software?

**Comment by postbird** [comment]
> > 
> 
> > @panoply Couldn't have said it better. It's @maraks right to do what he did. Anybody think AWS ever donated money to the colors.js project since they use it for their AWS CLI?! Nope...
> 
> I have no right to accuse @maraks of any of its practices. The main reason I use `chalk` is that we can avoid these potential risks by choosing items that are more standardized, more stable, and reviewed by multiple people.
> 
> Of course, I'll reserve my opinion, @maraks actions also hurt thousands of people who work for open source software. This kind of behavior will not change the behavior of those big companies like AWS, they have strong R&D capabilities, and "use doctrine" is just an incidental behavior. Even if we all shut down all open source software, it won't prompt them to donate. In the end it's still the people who participate and contribute to open source that get hurt.
> 
> Also, I think if the original intention of open source is for money, then why not sell closed source software?

![image](https://user-images.githubusercontent.com/16200574/148749450-ab29fb9a-468a-407a-b1d7-4f25b766c596.png)

**Comment by codingjlu** [comment]
I prefer(red) colors because chalk uses es6 imports, and it really messes things up in the middle of a big project :/

**Comment by panoply** [comment]
> The main reason I use chalk is that we can avoid these potential risks by choosing items that are more standardized, more stable, and reviewed by multiple people.

@postbird while that is the case, Sindre took it upon himself to move the vast majority of his packages to ESM. Imo (and many others) he jumped the gun on that and the entire commonjs/esm mess got a whole lot worse considering so many developers use his packages. I know of several folks who spent days having to fix projects that were operating atop his deps, got bumped and caused major issues.  I personally suffered at that otherwise despicable decision he thrust upon the node nexus when a single dep I used of his was updated by a junior dev and then the deployed to a AWS Lambda. Uncool stuff. 

Quite frankly, fuck Sindre and his packages. I'd argue that him jumping the gun on the ESM front is equally as disruptive as Maraks decision to delete his modules. Nonetheless they are his packages and like Marak he is free to do as he wishes with them but until support is viable I'll stick with commonjs packages or modules that support both ESM and CommonJS. Plus, https://github.com/lukeed/kleur is brilliant and Lukeed is a gifted developer.

**Comment by webdiscus** [comment]
@panoply, here are the most popular ANSI libraries for Node.js that I have found and used:

- [colorette](https://github.com/jorgebucaran/colorette)
- [picocolors](https://github.com/alexeyraspopov/picocolors)
- [cli-color](https://github.com/medikoo/cli-color)
- [color-cli](https://github.com/jaywcjlove/colors-cli)
- [ansi-colors](https://github.com/doowb/ansi-colors)
- [kleur](https://github.com/lukeed/kleur)
- [chalk](https://github.com/chalk/chalk)

and yet one alternative lib with standard API - [ansis](https://github.com/webdiscus/ansis) supports both CommonJs and ESM.

See [features comparison of all libs above](https://github.com/webdiscus/ansis#compare-most-popular-ansi-libraries).
See [features benchmark of all libs above](https://github.com/webdiscus/ansis#benchmark).
